sphinx/util/inspect.py

```diff
@@ -23,8 +25,25 @@
 memory_address_re = re.compile(r' at 0x[0-9a-f]{8,16}(?=>$)')
 
 
+def isgeneratorfunction(func):
+    """Like inspect.isgeneratorfunction but supports functools.partial."""
+    try:
+        if inspect.ismethod(func):
```

Comment:
I don't think the unpacking here is useful;  this should be handled at a higher level.  Because now, only generator functions get this (why not coroutines and others?)